test/testunusedvar.cpp

```diff
@@ -4149,6 +4150,96 @@ class TestUnusedVar : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void localvarmaybeunused() {
+        std::string passCases[] =
+        {
+R"(
+int main() {
+    [[maybe_unused]] int x;
+}
+)",
+R"(
+[[nodiscard]] int getX() { return 4; }
+int main() {
+    [[maybe_unused]] int x = getX();
+}
+)",
+R"(
+[[nodiscard]] int getX() { return 4; }
+int main() {
+    [[maybe_unused]] int x = getX();
+    x = getX();
+}
+)",
+R"(
+[[nodiscard]] int getX() { return 4; }
+int main() {
+    [[maybe_unused]] int x = getX();
+    x = getX();
+    std::cout << x;
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] const int x = getX();
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] const int& x = getX();
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] const int* x = getX();
+}
+)",
+R"(
+int main() {
+   [[maybe_unused]] int& x = getX();             
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] int* x = getX();
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] auto x = getX();
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] auto&& x = getX();
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] int x[] = getX();
+}
+)",
+R"(
+int main() {
+    [[maybe_unused]] constexpr volatile static int x = 1;
+}
+)",
+R"(
+[[maybe_unused]] inline int x = 1;
+)",
+R"(
+int main() {
+    [[maybe_unused]] [[anotherattribute]] const int* = 1;
+}
+)",
+        };
+        for (auto pass : passCases)
```

Comment:
I do not want to have a for loop like this. If there is a test failure then we can't see which specific case fails.